### DISCUSSION TITLE
How do I confirm I've compiled in SPIRAM support?

### DISCUSSION THREAD

**Discussion by BetterAutomations** [micropython/micropython#15156] [ESP32] [answered]
I have an ESP32-WROVER-IE with 8MB of SPIRAM. How can I confirm that the compilation I did was successful? ([Here were the steps I took to build](https://github.com/orgs/micropython/discussions/15131).) I see "Generic ESP32 module with SPIRAM with ESP32" in the bootup message, but that's just a string, and I want to ensure I did it right.

I guess gc.mem_free() and esp32.idf_heap_info() wouldn't show 4+MB if it was not successful, correct?

FYI to anyone who has the same question, [it is not currently possible without massive workarounds to use more than 4MB of SPIRAM on a generic ESP32](https://github.com/orgs/micropython/discussions/10078#discussioncomment-4237846).

```
MicroPython v1.23.0-preview.406.g30a9ccf4c.dirty on 2024-05-28; Generic ESP32 module with SPIRAM with ESP32
Type "help()" for more information.
>>> import gc
>>> import esp32
>>> gc.mem_free()
4183072
>>> esp32.idf_heap_info(esp32.HEAP_DATA)
[(32767, 32031, 31744, 32031), (4194304, 4134464, 4128768, 4134464), (240, 4, 0, 4), (7288, 4, 0, 4), (16648, 3592, 2560, 3100), (85904, 35872, 34816, 35872), (15072, 14656, 14336, 14656), (113840, 112968, 110592, 112968)]
>>>
```

**Comment by sosi-deadeye** [comment, marked as answer]
```print(f"{4183072 / 1024 ** 2:.2f} MiB")```

Output: `3.99 MiB`

Looks right. You can also test it with `x = bytearray(wanted_size)`,

**Comment by BetterAutomations** [comment]
TYVM
```
>>> x = bytearray(4000000)
>>>
```
